### ISSUE TITLE
Resume bots triângular nodeJS

### ISSUE THREAD

**Issue by potara69** [bmino/binance-triangle-arbitrage#104]
hello i'm from brazil and did tests with
tokio vps 32cpus ws + 50gb network
PING 15MS
I WILL POST THE PRINTS
the maximum power that normal people could have

THIS TYPE OF ARBITRATION IS NOT PROFITABLE, NOT IN BINANCE

MAY BE PROFITABLE TRIANGLE AND POCISIONED ARBITRATION BETWEEN EXCHANGES 3-4-6 EXCHANGES

or maybe a bot made in C # or C ++ that is faster than nodejs

![IMG_20200730_191911](https://user-images.githubusercontent.com/62059656/88980533-4dc87200-d29a-11ea-88c0-3ddda01565c5.jpg)

![IMG_20200730_192158](https://user-images.githubusercontent.com/62059656/88980509-45703700-d29a-11ea-9d43-0df4b247d751.jpg)
![IMG_20200730_192152](https://user-images.githubusercontent.com/62059656/88980512-4739fa80-d29a-11ea-842a-9c71a98f21e3.jpg)
![IMG_20200730_192134](https://user-images.githubusercontent.com/62059656/88980515-486b2780-d29a-11ea-8d1c-6dfa0d85b968.jpg)
![IMG_20200730_192038](https://user-images.githubusercontent.com/62059656/88980518-4903be00-d29a-11ea-981d-c4d199041cf4.jpg)
![IMG_20200730_192011](https://user-images.githubusercontent.com/62059656/88980520-4a34eb00-d29a-11ea-87eb-955c54d5428e.jpg)
![IMG_20200730_191927](https://user-images.githubusercontent.com/62059656/88980523-4b661800-d29a-11ea-9b5d-f8f826714751.jpg)
![IMG_20200730_191919](https://user-images.githubusercontent.com/62059656/88980529-4c974500-d29a-11ea-95e5-957e4f31ec88.jpg)

**Comment by potara69** [comment]
Awesome configs, FOR MINIMAL VALUE COINS IN TRADE.

{

    "KEYS": {
        "API": "",
        "SECRET": ""
    },

    "INVESTMENT": {
        "BASE": "USDT",
        "MIN": 12,
        "MAX": 24,
        "STEP": 1
    },

    "TRADING": {
        "ENABLED": true,
        "EXECUTION_STRATEGY": "linear",
        "EXECUTION_TEMPLATE": ["", "", ""],
        "EXECUTION_CAP": 0.075,
        "TAKER_FEE": 0.00,
        "PROFIT_THRESHOLD": 0.01,
        "AGE_THRESHOLD": 50,
        "WHITELIST": []
    },

    "HUD": {
        "ENABLED": false,
        "ARB_COUNT": 10
    },

    "LOG": {
        "LEVEL": "debug",
        "PRETTY_PRINT": true
    },

    "DEPTH": {
        "SIZE": 10,
        "PRUNE": false,
        "INITIALIZATION_INTERVAL": 75
    },

    "CALCULATION_COOLDOWN": 15

}

**Comment by bmino** [comment]
Perhaps a bot written in another language might be faster. Algorithmic trading in general and especially arbitrage is sooooo time sensitive. Simply lacking a direct line to the exchange might disqualify traders.

Thank you for post such in depth data! I've run the bot on a 32 core machine out of Tokyo last year to test out some websocket scaling issues and saw mixed results leading me to believe that processing power is not the limiting factor here. It appears that even in the single digits of millisecond latency, you need to be faster. I'm confident the bot is identifying legit opportunities but in many cases lacks the speed to execute in a timely manner.

A SIGNIFICANT performance improvement is running the bot in parallel execution mode which will effectively make the bot 3 times faster in executing positions but requires balances of more coins to be held. I haven't invested enough time and value to spread out an investment across enough coins to test this strategy

**Comment by potara69** [comment]
I only have 120usd, to test the parallel effectively I believe that a bankroll of 500 would be good, spreading 11-12 usd among the top 20 coin some profit, soon I will try to test it. difficult is to leave some money to invest in cryptos hahaha :(

**Comment by bmino** [comment]
I have been playing around with ~500usd and still think it's a pain to identify the coins and distribute a stack of funds equally amongst them. Each coin being used needs to have a sufficient balance equal to the max balance to be converted of the base currency.

Perhaps a nice feature would be an automated equal distribution of funds into the given whitelist to prep for running in parallel execution mode. ie you use 5 symbols in the whitelist and BTC as a base so the bot calculates if you have enough BTC to satisfy the minimum amount of each symbol and then distributes it out with BUY/SELL orders as appropriate

**Comment by potara69** [comment]
an idea you put in development the configuration support more currencies in just one instance

**Comment by bmino** [comment]
I'm not sure what you mean by that? If you mean having multiple "base currencies" this is possible and perhaps a great idea.

I'm gonna close this as there isn't a particular thread focused on. Feel free to open a new issue!
